### PR TITLE
Add UI support for management jobs in workflows

### PR DESCRIPTION
**Pull request by mabashian**
##### SUMMARY
link https://github.com/ansible/awx/issues/790

This has been supported via the API for a long time.  The tricky part was handling the `days` (days to keep) for some of the management jobs.  That value needs to flow in/out of extra_data so it's a bit different from other workflow node fields.

Here it is in action:

mgt_job_wf

I elected to just show the variables in the details view for simplicity.  If it's important we could pull `days` out of the object but more code brings added risk of bugs 🤷 .

Screen Shot 2021-07-01 at 11 23 33 AM

@wenottingham if you're not really into this as a feature that's fine - we can close this and reference it later.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
 - UI

### CODE REVIEW

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
Some things I found:
1) if you type/edit the days field it then shows up as a string in the variable details view after clicking on the `i` icon when hovering a node.  However, when you edit that same node, but don't edit the days field the number appears as an integer  in the variables detail after clicking on the `i` icon.  There doesn't seem to be any negative impact on functionality, just some ux inconsistency.

2) We don't seem to show the convergence data in the detail modal.  Should we?

3) `Days` vs `Days to keep`
Screen Shot 2021-07-06 at 10 19 09 AM
Screen Shot 2021-07-06 at 10 19 20 AM

4) Create a `Clean up job details` node and connect it to a `Cleanup expired sessions` node.  Then click on the `Cleanup expired sessions` node `i` icon.  Page crashes.
Screen Shot 2021-07-06 at 10 22 57 AM
Screen Shot 2021-07-06 at 10 23 08 AM

**Comment by mabashian** [comment]
> if you type/edit the days field it then shows up as a string in the variable details view after clicking on the i icon when hovering a node. However, when you edit that same node, but don't edit the days field the number appears as an integer in the variables detail after clicking on the i icon. There doesn't seem to be any negative impact on functionality, just some ux inconsistency.

Will include a fix for this

> We don't seem to show the convergence data in the detail modal. Should we?

Was probably just missed when we did the initial work.  Let's file an issue for this.

> Days vs Days to keep

The api expects the key to be `days` in `extra_data` so that can't change.  Labelling the field `Days` is ambiguous to me so I don't think we should change it to that.  Here's what the field looks like when you launch a management job:

Screen Shot 2021-07-12 at 12 01 24 PM

I'd propose changing the string to something like:

Screen Shot 2021-07-12 at 12 03 23 PM

so that it more closely matches the other interface.

> Create a Clean up job details node and connect it to a Cleanup expired sessions node. Then click on the Cleanup expired sessions node i icon. Page crashes.

I'll look at this

**Review by AlexSCorey — APPROVED** [review]
This looks good to me.  The unit tests pass, and the issues described earlier are resolved.  I'm triggering e2e tests now.

**Comment by tiagodread** [comment]
@mabashian could you rebase this one?

File: awx/ui_next/src/util/validators.js [hunk 1]
```diff
@@ -57,6 +57,12 @@ export function minLength(min) {
 
 export function minMaxValue(min, max) {
   return (value) => {
+    if (!Number.isFinite(min) && value > max) {
+      return t`This field must be a number and have a value less than ${max}`;
+    }
+    if (!Number.isFinite(max) && value < min) {
+      return t`This field must be a number and have a value greater than ${min}`;
```
Comment (tiagodread) [inline awx/ui_next/src/util/validators.js:64]: This is how the ${min} is being displayed.
 
image

**Review by tiagodread — APPROVED** [review]
Verified that mgmt jobs can be added as nodes to a workflows and run it, also tested the scheduled workflow with mgmt jobs.

**Pull request merged**